boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+
+#include <soc.h>
+#include <zephyr/drivers/gpio.h>
+
+#include <nrfx_gpiote.h>
+
+LOG_MODULE_REGISTER(nrf5340_audio_dk_nrf5340_cpuapp, CONFIG_LOG_DEFAULT_LEVEL);
+
+#if defined(CONFIG_BT_CTLR_DEBUG_PINS_CPUAPP)
+#include <../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h>
+#else
+#define DEBUG_SETUP()
+#endif
+
+static int core_config(void)
+{
+	int ret;
+	nrf_gpiote_latency_t latency;
+
+	latency = nrfx_gpiote_latency_get();
+
+	if (latency != NRF_GPIOTE_LATENCY_LOWPOWER) {
+		LOG_DBG("Setting gpiote latency to low power");
+		nrfx_gpiote_latency_set(NRF_GPIOTE_LATENCY_LOWPOWER);
+	}
+
+	/* USB port detection
+	 * See nPM1100 datasheet for more information
+	 */
+	static const struct gpio_dt_spec pmic_iset =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(nrf5340_audio_dk), pmic_iset_out_gpios);
+
+	if (!device_is_ready(pmic_iset.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure_dt(&pmic_iset, GPIO_OUTPUT_INACTIVE);
+	if (ret) {
+		return ret;
+	}
+
+	/* Set on-board DSP/HW codec as default */
+	static const struct gpio_dt_spec hw_codec_sel =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(nrf5340_audio_dk), hw_codec_sel_out_gpios);
+
+	if (!device_is_ready(hw_codec_sel.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENODEV;
+	}
+
+	/* Select on-board HW codec by default */
+	ret = gpio_pin_configure_dt(&hw_codec_sel, GPIO_OUTPUT_ACTIVE);
+	if (ret) {
+		return ret;
+	}
```

Comment:
These seem to go against the board porting guide https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#general-recommendations whilst one may say that's the only reason why someone would use this board (for audio applications) but the same thing applies to the nrf9160dk and having the modem disabled by default, you wouldn't use the nrf9160 unless you wanted a modem, but it is disabled by default to give a minimal interface as per the board porting guide.